zql/zql.peg

```diff
@@ -551,7 +551,7 @@ boomWord "boomWord" = chars:boomWordPart+ { RETURN(joinChars(chars)) }
 
 boomWordPart
   = "\\" s:escapeSequence  { RETURN(s) }
-  / !([\x00-\x1F\x5C(),!><=DQUOTE|SQUOTE;] / ws) . { RETURN(TEXT) }
+  / !ws . { RETURN(TEXT) }
```

Comment:
I think it would be better to define the syntax of a search word (what are the allowed characters and what are the escaped characters), then unescape the above before calling ReGgob.  This will help make it more easily documented.  Right now, it's hard to explain as anything that doesn't match something else parseable in the language.

Reply:
I reverted this change, I don't personally think this is making the language/grammar clearer but don't care enough to debate it.
I don't understand the comment about escaping though, this patch only adds `\=` and '\*`.  The equal sign is unescaped directly in the grammar rule, but we need to let reglob know which asterisks have been escaped so it can construct a correct regexp, I'm doing this by passing the backslash along, you seem to want to handle this in some different manner, but you haven't said exactly how...?